litellm/rerank_api/main.py

```diff
@@ -113,6 +117,8 @@ def rerank(  # noqa: PLR0915
             ProviderConfigManager.get_provider_rerank_config(
                 model=model,
                 provider=litellm.LlmProviders(_custom_llm_provider),
+                api_base=optional_params.api_base,
+                present_version_params=present_version_params
```

Comment:
can we avoid adding these two new params to `get_provider_rerank_config`. present_version seems specific to cohere and should be handled with CohereRerankConfig() and CohereRerankv2Config()

Reply:
We might still need it here because we need to use the params to determine whether we need to use the new Config or the old Config. These fields will also need to be used for Azure, if they start supporting the new client